core/src/client/mod.rs

```diff
@@ -0,0 +1,58 @@
+use derivative::Derivative;
+
+use crate::{config::Config, metadata::Metadata};
+
+/// Each client should be able to provide access to the following fields
+/// - runtime version
+/// - genesis hash
+/// - metadata
+
+#[derive(Derivative)]
+#[derivative(Debug(bound = ""), Clone(bound = ""))]
+pub struct ClientMetadata<C: Config> {
```

Comment:
not sure about the name `ClientMetadata` could be `ChainState`, `ClientState` or something.

Because it's not only metadata but runtime version and genesis hash....

Reply:
Yes, I am also not sure about the name. I named it `ClientBase` before. `ClientState` sounds good, any opinions on this @jsdw @lexnv ?